### PR TITLE
Handle IPv6 in bindAddr

### LINKED ISSUE #7633: net.bindAddr fails binding to all interfaces if `address == ""` for ipv6
```nim
import net
var socket = newSocket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP)
# socket.bindAddr(Port(1900), "::") # this compiles
socket.bindAddr(Port(1900))

bug.nim(4)               bug
net.nim(775)             bindAddr
oserr.nim(110)           raiseOSError
Error: unhandled exception: Invalid argument [OSError]
```

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Handle empty IPv4 and v6 correctly and add test. Closes #7633

### CODE REVIEW

File: lib/pure/net.nim [hunk 1]
```diff
@@ -755,21 +755,20 @@ proc bindAddr*(socket: Socket, port = Port(0), address = "") {.
   ## Binds ``address``:``port`` to the socket.
   ##
   ## If ``address`` is "" then ADDR_ANY will be bound.
+  var realaddr = address
+  if realaddr == "":
+    case socket.domain
+    of AF_INET6: realaddr = "::"
```
Comment (Varriount) [inline lib/pure/net.nim]: It kinda irks me that these strings will be created, just to be thrown away in the end. Ah well...

Comment (FedericoCeratto) [reply lib/pure/net.nim]: realaddr is being used... do you have any alternative?

Comment (Araq) [reply lib/pure/net.nim]: @Varriount with --gc:destructors these are pretty much zero cost.

**Comment by dom96** [comment]
LGTM

**Pull request merged**